components/brave_wallet/browser/swap_service_unittest.cc

```diff
@@ -322,4 +414,319 @@ TEST_F(SwapServiceUnitTest, IsSwapSupported) {
+  TestGetJupiterQuoteUint64Cases(R"(
+    {
+      "data": [
+        {
+          "inAmount": 10000,
+          "outAmount": 261273,
+          "amount": 10000,
+          "otherAmountThreshold": 258660,
+          "outAmountWithSlippage": 258660,
+          "swapMode": "ExactIn",
+          "priceImpactPct": 0.008955716118219659,
+          "marketInfos": [
+            {
+              "id": "2yNwARmTmc3NzYMETCZQjAE5GGCPgviH6hiBsxaeikTK",
+              "label": "Orca",
+              "inputMint": "EPjFWdd5AufqSSqeM2qN1xzybapC8G4wEGGkZwyTDt1v",
+              "outputMint": "MNDEFzGvMt87ueuHvVU9VcTqsAP5b3fTGPsHuuPA5ey",
+              "notEnoughLiquidity": false,
+              "inAmount": %s,
+              "outAmount": 117001203,
+              "priceImpactPct": 1.196568750220778e-7,
+              "lpFee": {
+                "amount": 30,
+                "mint": "EPjFWdd5AufqSSqeM2qN1xzybapC8G4wEGGkZwyTDt1v",
+                "pct": 0.003
+              },
+              "platformFee": {
+                "amount": 0,
+                "mint": "MNDEFzGvMt87ueuHvVU9VcTqsAP5b3fTGPsHuuPA5ey",
+                "pct": 0
+              }
+            }
+          ]
+        }
+      ],
+      "timeTaken": 0.044471802000089156
+    })");
+
+  // Test marketInfos->outAmount uint64 value
+  TestGetJupiterQuoteUint64Cases(R"(
+    {
+      "data": [
+        {
+          "inAmount": 10000,
+          "outAmount": 261273,
+          "amount": 10000,
+          "otherAmountThreshold": 258660,
+          "outAmountWithSlippage": 258660,
+          "swapMode": "ExactIn",
+          "priceImpactPct": 0.008955716118219659,
+          "marketInfos": [
+            {
+              "id": "2yNwARmTmc3NzYMETCZQjAE5GGCPgviH6hiBsxaeikTK",
+              "label": "Orca",
+              "inputMint": "EPjFWdd5AufqSSqeM2qN1xzybapC8G4wEGGkZwyTDt1v",
+              "outputMint": "MNDEFzGvMt87ueuHvVU9VcTqsAP5b3fTGPsHuuPA5ey",
+              "notEnoughLiquidity": false,
+              "inAmount": 10000,
+              "outAmount": %s,
+              "priceImpactPct": 1.196568750220778e-7,
+              "lpFee": {
+                "amount": 30,
+                "mint": "EPjFWdd5AufqSSqeM2qN1xzybapC8G4wEGGkZwyTDt1v",
+                "pct": 0.003
+              },
+              "platformFee": {
+                "amount": 0,
+                "mint": "MNDEFzGvMt87ueuHvVU9VcTqsAP5b3fTGPsHuuPA5ey",
+                "pct": 0
+              }
+            }
+          ]
+        }
+      ],
+      "timeTaken": 0.044471802000089156
+    })");
+
+  // lpFee->amount and platformFee->amount are not uint64 fields.
+  // Test cases can be found in swap_response_parser_unittest.cc.
+}
+
+TEST_F(SwapServiceUnitTest, GetJupiterSwapTransactions) {
+  SetInterceptor(R"(
+    {
+      "setupTransaction": "foo",
+      "swapTransaction": "bar",
+      "cleanupTransaction": "baz"
+    })");
+
+  bool callback_run = false;
+  swap_service_->GetJupiterSwapTransactions(
```

Comment:
Instead of callback_run + RunUntilIdle, move this to a member function of SwapServiceUnitTest and use `run_loop.Run/Quit` directly inside.